torchvision/prototype/datasets/_builtin/food101.py

```diff
@@ -83,9 +95,15 @@ def _make_datapipe(
 
         return Mapper(dp, self._prepare_sample)
 
-    def _generate_categories(self, root: Path) -> List[str]:
-        resources = self.resources(self.default_config)
-        dp = resources[0].load(root)
+    def _generate_categories(self) -> List[str]:
+        resources = self.resources()
+        dp = resources[0].load(self._root)
         dp = Filter(dp, path_comparator("name", "classes.txt"))
         dp = LineReader(dp, decode=True, return_path=False)
         return list(dp)
+
+    def __len__(self) -> int:
+        return {
+            "train": 75_750,
+            "test": 25_250,
+        }[self._split]
```

Comment:
Nit: do we always want to go with dicts here? For two values, I went for a simple `if` / `else`

```suggestion
        return 75_750 if self._split == "train" else 25_250
```

No strong opinion though.